whois/README.md

```diff
@@ -1,13 +1,120 @@
 # polkadot-whois
 
-Tracking of validators using on-chain and substrate telemetry data. The data can be queried by validator's `AccountId` or `(session_index, validator_index)`.
+A tool for converting and querying information about validators using interchangeable unique identifiers, information about a validator can be queried using the following indentifiers:
+- Validator index in `paraSessionInfo` pallet.
+- The account key.
+- The authority discovery id, provided in the hex format, it is the same format as what is displayed in `session.next_keys`.
+- The peer id, the network peer id used for connecting to other nodes.
 
-Usage:
+It prints a list of information for each queried validator, G.g:
+```
+validator_index=<INDEX<>>, account=<ACCOUNT>, peer_id=<PEER_ID>, authorithy_id_discover=0x<AUTHORITY_DISCOVERY>, addresses=<KNOWN_ADDRESSES>, version="POLKADOT_NODE_VERSION"
+```
+
+It uses the subp2p-explorer for querying the DHT information about each validator in the network.
+
+## Usage:
 
+### Query by validator index
+```
+cargo run  --bin polkadot-whois -- \
+ --ws=wss://rpc.polkadot.io:443 \
+ --genesis 0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3 \
+ --bootnodes /dns/polkadot-bootnode-0.polkadot.io/tcp/30333/p2p/12D3KooWSz8r2WyCdsfWHgPyvD8GKQdJ1UAiRmrcrs8sQB3fe2KU \
+ --address-format polkadot --timeout 900  --chain polkadot --session-index 9442 \
+ by-validator-index 295 293
```

Comment:
I ran into an error while attempting to execute it. I believe this occurred because the RPC node isn't an archive node and didn't respond to the previous query. Are you able to manage this kind of error?

Reply:
Done.